bin/ncp/SYSTEM/metrics.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+is_active() {
+  return 1
+}
+
+tmpl_metrics_enabled() {
+  (
+  . /usr/local/etc/library.sh
+  local param_active="$(find_app_param metrics.sh ACTIVE)"
+  [[ "$param_active" == yes ]] || exit 1
+  )
+}
+
+install() {
+
+  # Subshell to return on failure  instead of exiting (due to set -e)
+  (
+
+  set -e
+  cat > /etc/default/prometheus-node-exporter <<'EOF'
+ARGS="--collector.filesystem.ignored-mount-points=\"^/(dev|proc|run|sys|mnt|var/log|var/lib/docker)($|/)\""
+EOF
+  apt-get update && apt-get install -y --no-install-recommends prometheus-node-exporter
+
+  # TODO: Docker support?
+  systemctl disable prometheus-node-exporter
+  service prometheus-node-exporter stop
+
+  )
+}
+
+configure() {
+
+  if [[ "$ACTIVE" != yes ]]
+  then
+    bash /usr/local/etc/ncp-templates/nextcloud.conf.sh --defaults > /etc/apache2/sites-available/nextcloud.conf
+
+    systemctl disable prometheus-node-exporter
+    service prometheus-node-exporter stop
+  else
+    [[ -n "$USER" ]] || {
+      echo "ERROR: User can not be empty!" >&2
+      return 1
+    }
+
+    [[ -n "$PASSWORD" ]] || {
+      echo "ERROR: Password can not be empty!" >&2
+      return 1
+    }
+
+    [[ ${#PASSWORD} -ge 10 ]] || {
+      echo "ERROR: Password must be at least 10 characters long!" >&2
+      return 1
+    }
+
+    local htpasswd_file="/usr/local/etc/metrics.htpasswd"
+    [[ -f "${htpasswd_file}" ]] && rm "${htpasswd_file}"
+    echo "$PASSWORD" | htpasswd -ciB "${htpasswd_file}" metrics
+
+    bash /usr/local/etc/ncp-templates/nextcloud.conf.sh > /etc/apache2/sites-available/nextcloud.conf || {
+      echo "An unexpected error occurred while configuring apache. Rolling back..." >&2
+      bash /usr/local/etc/ncp-templates/nextcloud.conf.sh --defaults > /etc/apache2/sites-available/nextcloud.conf
+      return 1
+    }
+
+    systemctl enable prometheus-node-exporter
+    service prometheus-node-exporter start
+
+    echo "Metric endpoint enabled. You can test it at https://nextcloudpi.local/metrics/system (or under your NC domain under the same path)"
+  fi
+  service apache2 reload
```

Comment:
```suggestion
bash -c "sleep 2 && service apache2 reload" &>/dev/null &
```
same reason as below

Reply:
Hm, I believe I tested that... Reload seems to work. But I'll change it nonetheless

Reply:
if the server shuts down before sending back the HTTP response, we'll get the red ball and an error message on ncp-web